src/Tools/AnalyzerRunner/Program.cs

```diff
@@ -0,0 +1,480 @@
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Diagnostics.Telemetry;
+using Microsoft.CodeAnalysis.MSBuild;
+using File = System.IO.File;
+using Path = System.IO.Path;
+
+namespace AnalyzerRunner
+{
+    /// <summary>
+    /// AnalyzerRunner is a tool that will analyze a solution, find diagnostics in it and will print out the number of
+    /// diagnostics it could find. This is useful to easily test performance without having the overhead of visual
+    /// studio running.
+    /// </summary>
+    class Program
+    {
+        public static async Task Main(string[] args)
+        {
+            var options = Options.Create(args);
+            if (options == null)
+            {
+                return;
+            }
+
+            CancellationTokenSource cts = new CancellationTokenSource();
+            Console.CancelKeyPress +=
+                (sender, e) =>
+                {
+                    e.Cancel = true;
+                    cts.Cancel();
+                };
+
+            var analyzerAssemblies = GetAnalyzerAssemblies(options.AnalyzerPath);
+            var analyzers = GetAllAnalyzers(analyzerAssemblies);
+            analyzers = FilterAnalyzers(analyzers, options).ToImmutableArray();
+
+            if (analyzers.Length == 0)
+            {
+                Utilities.WriteLine("No analyzers found", ConsoleColor.Red);
+                Utilities.PrintHelp();
+                return;
+            }
+            var cancellationToken = cts.Token;
+
+            Stopwatch stopwatch = Stopwatch.StartNew();
+            using (MSBuildWorkspace workspace = MSBuildWorkspace.Create())
+            {
+                Solution solution = await workspace.OpenSolutionAsync(options.SolutionPath, cancellationToken).ConfigureAwait(false);
+                bool proceedLookingForAnalyzerReferences = true;
+                while (proceedLookingForAnalyzerReferences)
+                {
+                    proceedLookingForAnalyzerReferences = false;
+                    foreach (var project in solution.Projects)
+                    {
+                        if (project.AnalyzerReferences.Any())
+                        {
+                            proceedLookingForAnalyzerReferences = true;
+                            solution = project.WithAnalyzerReferences(ImmutableArray<AnalyzerReference>.Empty).Solution;
+                        }
+                    }
+                }
+
+                Console.WriteLine($"Loaded solution in {stopwatch.ElapsedMilliseconds}ms");
```

Comment:
this should at least print out how many projects, files are there or etc. to have poor main's check to see MSWorkspace.OpenSolution worked properly.

and I do think we should have options to check msbuildworkspace.opensolution worked properly with GetDiagnostics. one should always try that option when trying new solution to make sure it loaded properly.